tests/test_template.py

```diff
@@ -38,3 +39,16 @@ def test_find_shapes_from_one_slide(template):
 def test_find_shapes_index_out_of_range(template):
     with pytest.raises(IndexError):
         shapes = template._find_shapes(0, 'logo')
+
+def test_get_all_shapes(template):
+    shapes = template._get_all_shapes()
+    for shape in shapes:
+        assert str(shape.text) and str(shape.name)
```

Comment:
So how does this test that I 'get all the shapes'? I'm afraid that you'd have to mock the `_presentation` to verify this...

Reply:
I simply named the test after the function it is testing. But no, it is not actually testing that I have all the shapes. But code to test this is necessarily going to be as complex as the code that performs it.

At the very least, the code checks that what was obtained has at least some properties we would expect to find in slides :)

Reply:
Yes, the test would be rather complex. At least you'd have to monkey patch the `Presentation` class (because we don't use dependency injection) and configure the mock (which seems tedious). Further, there are people out there that say "don't test private methods, just test the public interface", i.e., you would not write a test for `_get_all_shapes()` at all, but only test it indirectly by testing the public method it is used in.

Would probably the simplest to just give the test a more descriptive (cool slang for "longer") name ;-)